src/typed_array.rs

```diff
@@ -0,0 +1,360 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::{sys, JSException, JSObject, JSTypedArray, JSTypedArrayType, JSValue};
+use std::{ptr, slice};
+
+impl JSTypedArray {
+    /// Create a new [`Self`] from its raw pointer directly.
+    ///
+    /// # Safety
+    ///
+    /// Ensure `raw` is valid.
```

Comment:
Maybe also say it must be a typed array? Or can it reasonably be something else?

Reply:
I've updated the doc to say it must be a typed array, but the C API is defensive: if the `JSObjectRef` isn't a typed array, it shouldn't crash. But still, it must be a `JSObjectRef` of a typed array.